### PR TITLE
create Address model for Location model. add address in farmerDetails…

### PR DESCRIPTION
**Pull request by YvesC88**
I formatted and added the address to the FarmerDetailsView. I also structured the address by creating an Address model.

I'm not sure about the FarmerPlace extension in MapViewModel for the title and name of the image for the Marker because I also need for the FarmerDetailsView

### CODE REVIEW

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift [hunk 1]
```diff
@@ -36,20 +37,22 @@ struct FarmerDetailsView: View {
                         .clipShape(.rect(cornerRadius: 16))
                         .shadow(radius: 2)
                         .allowsHitTesting(false)
```
Comment (MathisDetourbet) [inline MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: Why did you add this modifier?

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift [hunk 2]
```diff
@@ -36,20 +37,22 @@ struct FarmerDetailsView: View {
                         .clipShape(.rect(cornerRadius: 16))
```
Comment (MathisDetourbet) [inline MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: Why you didn't add these modifiers in the property declaration?

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift [hunk 3]
```diff
@@ -36,20 +37,22 @@ struct FarmerDetailsView: View {
...
                         .shadow(radius: 2)
                         .allowsHitTesting(false)
+                    addressSection
+                        .background(.thinMaterial)
+                        .clipShape(.rect(cornerRadius: 20))
```
Comment (MathisDetourbet) [inline MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: Same here? Why not adding that in address section instead?

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift [hunk 4]
```diff
@@ -114,17 +117,42 @@ Bienvenue à la ferme de William, où la nature prospère en harmonie. William c
         }
         .mapControlVisibility(.hidden)
     }
+
+    private var directionButton: some View {
+        Button {
+            #warning("Apple Plan")
```
Comment (MathisDetourbet) [inline MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: Show Apple plan is clearer

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift [hunk 5]
```diff
@@ -114,17 +117,42 @@ Bienvenue à la ferme de William, où la nature prospère en harmonie. William c
...
+        } label: {
+            Text("Y aller")
```
Comment (MathisDetourbet) [inline MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift:122]: The text could have been in the viewModel

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift [hunk 6]
```diff
@@ -114,17 +117,42 @@ Bienvenue à la ferme de William, où la nature prospère en harmonie. William c
...
+        }
+        .font(.headline)
+    }
+
+    private var addressSection: some View {
+        HStack {
+            VStack {
```
Comment (MathisDetourbet) [inline MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: Why do you use a VStack here? You have only one element inside

File: MonMaraicher/Model/Address.swift [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Yves Charpentier on 05/03/2024.
+//
+
+import Foundation
+
+struct Address: Hashable {
+    let streetNumber: Int?
+    let streetName: String
+    let zip: Int
```
Comment (MathisDetourbet) [inline MonMaraicher/Model/Address.swift]: zipCode in English ;)

File: MonMaraicher/Model/FarmerPlace.swift [hunk 8]
```diff
@@ -13,27 +13,97 @@ struct FarmerPlace: Identifiable, Hashable {
...
-                      FarmerPlace(name: "Chez William", location: FarmerPlace.chezWilliam, imageNames: ["farmer1", "farmer2", "farmer3", "farmer4", "farmer5", "farmer6"]),
-                      FarmerPlace(name: "Maraîchers Bio de Grabels", location: FarmerPlace.bio, imageNames: ["farmer1", "farmer2", "farmer3", "farmer4", "farmer5", "farmer6"]),
-                      FarmerPlace(name: "Fred producteur", location: FarmerPlace.fred, imageNames: ["farmer1", "farmer2", "farmer3", "farmer4", "farmer5", "farmer6"]),
-                      FarmerPlace(name: "Mon Maraîcher", location: FarmerPlace.maraicher, imageNames: ["farmer1", "farmer2", "farmer3", "farmer4", "farmer5", "farmer6"]),
-                      FarmerPlace(name: "La ferme du Chot", location: FarmerPlace.chot, imageNames: ["farmer1", "farmer2", "farmer3", "farmer4", "farmer5", "farmer6"])
+    static let all = [FarmerPlace(name: "chez william", location: FarmerPlace.applePark1, imageNames: ["farmer1", "farmer2", "farmer3", "farmer4", "farmer5", "farmer6"]),
```
Comment (MathisDetourbet) [inline MonMaraicher/Model/FarmerPlace.swift:16]: The file is becoming huge.. I would move all static inside another file to avoid mixing your data with the model declaration. It's up to you ;)

File: MonMaraicher/Model/FarmerPlace.swift [hunk 9]
```diff
@@ -13,27 +13,97 @@ struct FarmerPlace: Identifiable, Hashable {
...
+    )
+    private static let applePark4 = Location(
+        latitude: 37.32238963337884,
+        longitude: -121.98463768108974,
+        address: Address(streetNumber: 139,
+                         streetName: "rue de la rauze",
+                         zip: 34070,
+                         city: "montpellier")
+    )
+    private static let applePark5 = Location(
```
Comment (MathisDetourbet) [inline MonMaraicher/Model/FarmerPlace.swift:101]: You could even write all these data inside a json file you load at application start. When you will have a backend you will need to do that

Comment (MathisDetourbet) [reply MonMaraicher/Model/FarmerPlace.swift:101]: Let it like that for now it's only suggestion for the future ;)

**Review by MathisDetourbet — APPROVED** [review]
Some small suggestions. Nicely done :)

**Comment by MathisDetourbet** [comment]
If you also need the title and the image name in another view it is a good argument to create a FarmerPlaceViewModel ;)

**Comment by YvesC88** [comment]
So that means that I would have two references of two viewmodels for a single view. Isn’t it embarrassing?

**Comment by MathisDetourbet** [comment]
Absolutely not

**Comment by MathisDetourbet** [comment]
You could even move your marker component into a new swiftUI view and that view use the farmerplace viewModel

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift (hunk 2 above)
Comment (YvesC88) [reply MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: I wanted to keep the most important changes here. But your suggestion is not bad

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift (hunk 1 above)
Comment (YvesC88) [reply MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: To avoid having a bad experience when the user scrolls and touches the mini map

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift (hunk 3 above)
Comment (YvesC88) [reply MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: It's true

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift (hunk 4 above)
Comment (YvesC88) [reply MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: Agree with you, it was an oversight on my part

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift (hunk 6 above)
Comment (YvesC88) [reply MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: Yes, bad choice in the use of stacks

File: MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift (hunk 1 above)
Comment (MathisDetourbet) [reply MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift]: Ok makes sense

**Comment by YvesC88** [comment]
I can't understand the concept of the FarmerPlaceViewModel and implement it. I also can't visualize the technique for a view for the Marker without depending on the Map component and without complicating the project.
So I would like to talk with you about these topics

**Comment by MathisDetourbet** [comment]
Ok no pb. But I think you can merge the PR like that. It is good enough ;)
